toll-schema/src/main/java/global/maplink/toll/schema/request/LegRequest.java

```diff
@@ -1,22 +1,27 @@
 package global.maplink.toll.schema.request;
 
+import global.maplink.toll.schema.Condition;
 import global.maplink.toll.schema.Coordinates;
 import global.maplink.toll.schema.TollVehicleType;
 import lombok.*;
 
+import java.time.Instant;
 import java.util.ArrayList;
 import java.util.List;
 
-import static java.util.Collections.emptyList;
-
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
 @Builder
 public class LegRequest {
 
+    @Builder.Default
+    private Long calculationDate = Instant.now().toEpochMilli();
+
     private TollVehicleType vehicleType;
 
+    private Condition condition = new Condition();
```

Comment:
condition pode ser null né, assim não envia o objeto vazio

Reply:
Se `condition` for null, ele usará os valores padrão definidos no objeto `Condition`. Isso garante que ele sempre tenha valores consistentes.

Reply:
não deveria né, condition null deveria ser null, se o usuario não especifica condições ele não deveria ter um valor default

Reply:
Condition representa o `billingType` e o `period`. Se ambos vierem como `null`, isso pode impactar o algoritmo que o @felipemendes1994  está implementando, não é? Além disso, atualmente, na `Toll`, o próprio `billing` já possui um valor default. Estamos apenas adicionando um valor default ao `period` e centralizando essa regra em um único objeto.

Reply:
a condition não pode ter valor default
o comportamento é, na ausência de conditions, a toll se comporta como hoje